tests/Unit/Evolution/Systems/Cce/Actions/Test_ScriObserveInterpolated.cpp

```diff
@@ -185,124 +187,40 @@ struct test_metavariables {
-    const double time, Tags::ScriPlus<Tags::Psi1> /*meta*/) noexcept {
-  const auto time_factor = (1.0 + linear_coefficient * (time) +
-                            quadratic_coefficient * square(time));
-  const auto expected_psi_2 =
-      get(get<Tags::ScriPlus<Tags::Psi2>>(random_values)).data() * time_factor;
-  const auto expected_psi_3 =
-      get(get<Tags::ScriPlus<Tags::Psi3>>(random_values)).data() * time_factor;
-  const auto expected_psi_4 =
-      get(get<Tags::TimeIntegral<Tags::ScriPlus<Tags::Psi4>>>(random_values))
-          .data() *
-      (linear_coefficient + 2.0 * quadratic_coefficient * time);
-  const auto eth_u =
-      get(get<Tags::EthInertialRetardedTime>(random_values)).data() *
-      time_factor;
-  return get(get<Tags::ScriPlus<Tags::Psi1>>(random_values)).data() *
-             time_factor +
-         1.5 * expected_psi_2 * eth_u + 0.75 * expected_psi_3 * square(eth_u) +
-         0.125 * expected_psi_4 * pow<3>(eth_u);
-}
-
-ComplexDataVector compute_expected_field(
-    const Variables<typename test_metavariables::cce_scri_tags>& random_values,
-    const double linear_coefficient, const double quadratic_coefficient,
-    const double time, Tags::ScriPlus<Tags::Psi0> /*meta*/) noexcept {
-  const auto time_factor = (1.0 + linear_coefficient * (time) +
-                            quadratic_coefficient * square(time));
-  const auto expected_psi_1 =
-      get(get<Tags::ScriPlus<Tags::Psi1>>(random_values)).data() * time_factor;
-  const auto expected_psi_2 =
-      get(get<Tags::ScriPlus<Tags::Psi2>>(random_values)).data() * time_factor;
-  const auto expected_psi_3 =
-      get(get<Tags::ScriPlus<Tags::Psi3>>(random_values)).data() * time_factor;
-  const auto expected_psi_4 =
-      get(get<Tags::TimeIntegral<Tags::ScriPlus<Tags::Psi4>>>(random_values))
-          .data() *
-      (linear_coefficient + 2.0 * quadratic_coefficient * time);
-  const auto eth_u =
-      get(get<Tags::EthInertialRetardedTime>(random_values)).data() *
-      time_factor;
-  return get(get<Tags::ScriPlus<Tags::Psi0>>(random_values)).data() *
-             time_factor +
-         2.0 * expected_psi_1 * eth_u + 0.75 * expected_psi_2 * square(eth_u) +
-         0.5 * expected_psi_3 * pow<3>(eth_u) +
-         0.0625 * expected_psi_4 * pow<4>(eth_u);
+    const double time, TagToCalculate /*meta*/) noexcept {
+  const size_t size = random_values.number_of_grid_points();
+  Scalar<DataVector> linear_coefficient_vector;
+  get(linear_coefficient_vector) = DataVector{size, linear_coefficient};
+  Scalar<DataVector> quadratic_coefficient_vector;
+  get(quadratic_coefficient_vector) = DataVector{size, quadratic_coefficient};
+  Scalar<DataVector> time_vector;
+  get(time_vector) = DataVector{size, time};
+  std::string tag_name = db::tag_name<TagToCalculate>();
+  for(auto& char : tag_name) {
+    if (char == '(' or char == ')') {
+      char = '_';
+    }
+  }
+  while(tag_name[tag_name.size() - 1] == '_') {
```

Comment:
You may be able to use Boost's trimming https://www.boost.org/doc/libs/1_73_0/doc/html/string_algo/reference.html#header.boost.algorithm.string.trim_hpp to achieve this, or just `tag_name.erase(tag_name.find_last_not_of('_'))`